routes/variableurl.go

```diff
@@ -0,0 +1,123 @@
+package routes
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	parseIndex uint
+
+	ArgName string
+
+	ParseInt struct {
+		ArgName ArgName
+	}
+	ParseFloat32 struct {
+		ArgName ArgName
+	}
+	ParseFloat64 struct {
+		ArgName ArgName
+	}
+	ParseString struct {
+		ArgName ArgName
+	}
+	ParseVoid struct {
+		Required string
+	}
+	ParseEnd struct{}
+)
+
+const (
+	index = parseIndex(iota)
+)
+
+var (
+	badURLTypeMessage  = ErrorBody{"Bad URL type."}
+	badURLTypeCode     = 400
+	notFoundURLMessage = ErrorBody{"Not Found."}
+	notFoundURLCode    = 404
+)
+
+func incrementIndex(a Arguments) int {
```

Comment:
I was surprised when I realised later in this file that `incrementIndex` returns something : the name doesn’t reflect that at all. You shouldn’t write functions which both _do_ something and _give you_ something unless it is clear what they do and return.